spec/presenters/homepage_presenter_spec.rb

```diff
@@ -176,7 +176,8 @@
           let(:startdate) { date_last_year }
 
           it "exposes the correct data for the data box" do
-            create_list(type, expected_count, status, assigned_to: user, startdate:)
+            logs = build_list(type, expected_count, status, assigned_to: user, startdate:)
+            logs.each { |log| log.save(validate: false) }
```

Comment:
do we think this is acceptable? it feels a little iffy to me to skip all validations when all I really want to do is skip the date validation. But it would start getting quite verbose if I do create_list, each, update date only, save(validate: false)
One reason I was kind of fine with this in this case is we create_list with everything the same other than startdate elsewhere in the file, but there's nothing to ensure that that is always so.

Reply:
We used to have a feature flag for date validations, if we bring it back we could set it here

Otherwise we could create the logs and then update the dates and only skip the validations on update, but I get how that could be problematic

Another options could be adding something similar like `skip_dpo_validation` but idk if I love that

Reply:
We've got an `ignore_validation_errors` trait for lettings logs that might make this a bit nicer, although still has the problem of ignoring everything - would be very up for a trait that just skipped date validation.

Reply:
hmmm. I can't immediately think of a way of doing that due to deriving clearing etc
if you tried to save it with the date as nil, then everything after setup would be cleared
if you saved with today's date, then updated it without validation, then you could get different values derived on the first save which could be very confusing to debug one day.

Reply:
actually I will have a quick go to see if you can use TimeCop in the factory

Reply:
it is possible, and it works fine in conjunction with in progress logs on first pass, but ti doesn't play nicely with the completed trait, so I'm not sure if it isn't just going to introduce more confusion when working with tests, open to thoughts. For reference the code was:
```
    trait :skip_startdate_validation do
      to_create do |log|
        Timecop.freeze(log.startdate) do
          log.save!
        end
      end
    end

```